### PR TITLE
Database migration: updated scripts and migration fallback

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
- re-create database if migration is not possible (from version < 10)
- proper migration version range support
- migration scripts updated
- additional migration tests

### CODE REVIEW

File: golem/database/database.py [hunk 1]
```diff
@@ -43,14 +43,23 @@ def get_user_version(self) -> int:
     def set_user_version(self, version: int) -> None:
         self.db.execute_sql('PRAGMA user_version = {}'.format(version))
 
-    def _create_database(self) -> None:
+    def _drop_tables(self):
+        log.info("Removing database")
```
Comment (maaktweluit) [inline golem/database/database.py]: Should this message also be renamed to something like "removing tables"?

**Review by maaktweluit — APPROVED** [review]
LGTM!

One tiny comment

File: golem/database/database.py (hunk 1 above)
Comment (mfranciszkiewicz) [reply golem/database/database.py]: Oh yes, I missed that, thanks

File: golem/database/migration/migrate.py [hunk 2]
```diff
@@ -1,9 +1,23 @@
+import logging
+
+from peewee import DatabaseError
 from peewee_migrate import Migrator
 
 from golem.database.migration import default_migrate_dir, patch_peewee
 from golem.database.migration.router import Router
 
 
+log = logging.getLogger('golem.db')
```
Comment (jiivan) [inline golem/database/migration/migrate.py]: `logger` is more popular in golem

**Review by jiivan — APPROVED** [review]
lgtm

**Pull request merged**